staging/src/k8s.io/apiserver/pkg/server/config_test.go

```diff
@@ -32,6 +32,22 @@ import (
 	"k8s.io/client-go/rest"
 )
 
+func TestAuthorizeClientBearerToken(t *testing.T) {
+	// All of these should do nothing (not panic)
+	AuthorizeClientBearerToken(nil, nil, nil)
```

Comment:
Might be easier to reason about if you use 3 for loops instead of exploding the Cartesian product.

Reply:
It will be more lines of code, and I thought the pattern was obvious enough, but the nested loops are preferred then I will do that.